### DISCUSSION TITLE
thumbnails:  create thumbnails if missing

### DISCUSSION THREAD

**Discussion by centralhardware** [photoprism/photoprism#2556] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

if thumbnails file is missing black rectangle showing.
**Describe the solution you'd like**

create thumbnails right now if the file is lost

same as [Dynamic Previews](https://docs.photoprism.app/user-guide/settings/advanced/#dynamic-previews) but cache thumbnail

**Comment by graciousgrey** [comment]
Thumbnails should not be missing unless they have been manually deleted from the cache folder. 
You can create missing ones using the thumbs command.

**Comment by lastzero** [comment, marked as answer]
Note dynamically created thumbnails will be cached. So the feature you are requesting is already implemented. In addition you can re-index or run the thumbs command in a terminal to generate missing thumbnails.

**Reply by centralhardware** [reply]
thankyou

**Comment by heberuriegas** [comment]
Same issue here I have > 23k photos and hq videos and I had missing thumbnails everywhere.

After enable --thumb-uncached option (on-demand creation of missing thumbnails) it works but the thumbnails has low quality.

**Reply by lastzero** [reply]
Anything in your logs? Did you change settings, so a previous configuration might have prevented proper thumbnails generation? What version are you using?

**Reply by heberuriegas** [reply]
Yes I have several. I'm going to paste here only the relevant ones.

```
index: failed creating thumbnails for 2020/03/20200306_205728_B1865BD3.jpg (EOF while decoding)

index: failed adding main jpg file 2020/03/20200306_205728_B1865BD3.jpg

index: failed creating thumbnails for 2014/09/20140915_112235_8C1A892A.jpg (invalid JPEG format: short Huffman data while decoding)

index: failed creating thumbnails for 2016/04/20160412_150020_BF2488A7.jpg (invalid JPEG format: bad Huffman code while decoding)

thumb: thumbnail not cached
```

**Reply by heberuriegas** [reply]
Also this happen when I ran the thumbs command

```
docker-compose exec photoprism photoprism thumbs
INFO[2022-07-28T09:27:27Z] PhotoPrism® needs your support!              
INFO[2022-07-28T09:27:27Z] Visit https://photoprism.app/membership to learn more. 
ERRO[2022-07-28T09:35:58Z] resample: invalid JPEG format: bad Huffman code while decoding
```

**Reply by lastzero** [reply]
I don't see how the `--thumb-uncached` option would change that? Seems these files can't be read in general.

**Reply by heberuriegas** [reply]
I'm not sure about that because I can see every picture and video even without thumbnails.

I'm going to keep looking what is going on and let you know if I can find something interesting.

Awesome job with photoprism it is my family life album.
